File: 04/4.py
def main ():
	largest = 0 
	for i in range(999 , 99 , -1):
		for b in range(i , 99 , -1):
			tmp = i * b
			if (str(tmp)[::-1] == str(tmp) and tmp > largest):
				largest = tmp
				break
	print(largest)


main()

File: 04/4.rb
def main()
	largest = 0
	for i in 999.downto(100)
		for a in i.downto(100)
			tmp = a * i
			if ("#{tmp}".reverse == "#{tmp}" && tmp > largest )
				largest  = tmp
				break
			end
		end
	end
	puts largest
end

if __FILE__ == $0
	main()
end
